lib/auth/authtest/authtest.go

```diff
@@ -944,6 +967,10 @@ func NewTestTLSServer(cfg TLSServerConfig) (*TLSServer, error) {
 		return nil, trace.Wrap(err)
 	}
 
+	if err := srv.setupInventoryCache(); err != nil {
+		return nil, trace.Wrap(err)
+	}
```

Comment:
I think this is the wrong layer to instantiate the inventory cache. We can move this to (InitAuthCache)[https://github.com/gravitational/teleport/blob/fcc6b798afa5f39f9353129ee8c0dafcc3611593/lib/auth/authtest/authtest.go#L536-L596] which should be called whenever a test auth instance is created with the cache enabled. Alternatively, we could update `AuthCacheParams` to indicate whether the inventory cache should be created there.